### PR TITLE
FEAT-#3724: Converting self.resample_args for Resampler into a dict

### LINKED ISSUE #3724: Converting self.resample_args for Resampler into a dict
self.resample_args for Resampler should be converted into a dict to ensure simplicity of handling resample parameters at the query compiler level.

### PR DESCRIPTION
**Pull request by Rubtsowa**
Signed-off-by: Maria Rubtsova <maria.rubtsova@intel.com>

## What do these changes do?
self.resample_args for Resampler converted into a dict to ensure simplicity of handling resample parameters at the query compiler level.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3724 
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -3111,21 +3111,21 @@ def __init__(
...
-            kind,
-            loffset,
-            base,
-            on,
-            level,
-            origin,
-            offset,
-        ]
-        self.__groups = self.__get_groups(*self.resample_args)
+        self.resample_args = {
```
Comment (vnlitvinov) [inline modin/pandas/base.py:3114]: pls rename the attribute to be consistent to `self.resample_kwargs`

File: modin/pandas/base.py [hunk 2]
```diff
@@ -3144,9 +3144,7 @@ def __getitem__(self, key):
         """
 
         def _get_new_resampler(key):
-            subset = self._dataframe[key]
-            resampler = type(self)(subset, *self.resample_args)
-            return resampler
+            return Resampler(self._dataframe[key], **self.resample_args)
```
Comment (vnlitvinov) [inline modin/pandas/base.py:3147]: I think @YarShev was against my simplification - this indeed makes it harder to make a subclass if we create `Resampler` explicitly

**Review by vnlitvinov — CHANGES_REQUESTED** [review]

File: modin/pandas/base.py [hunk 3]
```diff
@@ -3111,21 +3111,21 @@ def __init__(
...
-            kind,
-            loffset,
-            base,
-            on,
-            level,
-            origin,
-            offset,
-        ]
-        self.__groups = self.__get_groups(*self.resample_args)
+        self.resample_kwargs = {
```
Comment (YarShev) [inline modin/pandas/base.py:3114]: The comment above should be removed.

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Comment by YarShev** [comment]
@Rubtsowa , we should probably add the changes to BaseQueryCompiler too since it seems the CI job failed because of that.

**Comment by vnlitvinov** [comment]
> @Rubtsowa , we should probably add the changes to BaseQueryCompiler too since it seems the CI job failed because of that.

I don't see any CI failures here, what are you talking about?

**Review by vnlitvinov — APPROVED** [review]
LGTM

**Comment by YarShev** [comment]
All entries `resample_args : list` -> `resample_kwargs : dict` need to be replaced in base QC.

**Review by vnlitvinov — APPROVED** [review]
LGTM assuming CI passes!

**Comment by vnlitvinov** [comment]
> All entries `resample_args : list` -> `resample_kwargs : dict` need to be replaced in base QC.

Good catch!

**Review by YarShev — APPROVED** [review]

**Pull request merged**